src/cli/cli.cpp

```diff
@@ -3758,9 +3762,241 @@ void Interpreter::HandleDiagnosticGetResponse(Message &aMessage, const Ip6::Mess
 
     mServer->OutputFormat("\r\n");
 
-    AppendResult(OT_ERROR_NONE);
+    // Output Network Diagnostic TLV values in standard YAML format.
+    while ((error = otThreadGetNextDiagnosticTlv(&aMessage, &iterator, &diagTlv)) == OT_ERROR_NONE)
+    {
+        size_t column = 0;
```

Comment:
What was the reasoning to use `size_t` type?

Reply:
going to use `uint16_t` for the column (`uint8_t` has the risk of overflow).